### PR TITLE
🎮 Added Discord Rich Presence

### LINKED ISSUE #2332: Discord rich presence
Research and implement: https://discordapp.com/developers/docs/rich-presence/how-to

### PR DESCRIPTION
**Pull request by AnotherFoxGuy**
Fixes #2332

### CODE REVIEW

**Comment by tritonas00** [comment]
Great!

A cmake option to turn it off would be nice

**Comment by CuriousMike56** [comment]
Maybe also add a in-game setting for those that don't want it? And I think the terrain name (e.g. Auriga Proving Grounds instead of 31.terrn2) should be shown instead.

**Comment by Zentro** [comment]
Rich presence can be disabled within Discord itself I'm pretty sure.

File: source/main/utils/Utils.cpp [hunk 1]
```diff
@@ -37,6 +39,54 @@
...
 
+void InitDiscord()
+{
+    DiscordEventHandlers handlers;
+    memset(&handlers, 0, sizeof(handlers));
+    handlers.ready = handleDiscordReady;
+    handlers.errored = handleDiscordError;
+
+    // Discord_Initialize(const char* applicationId, DiscordEventHandlers* handlers, int autoRegister, const char* optionalSteamId)
+    Discord_Initialize("492484203435393035", &handlers, 1, "1234");
```
Comment (only-a-ptr) [inline source/main/utils/Utils.cpp]: Where does the appID come from?

**Comment by only-a-ptr** [comment]
> @CuriousMike56 Maybe also add a in-game setting for those that don't want it? And I think the terrain name (e.g. Auriga Proving Grounds instead of 31.terrn2) should be shown instead.

Both seconded

File: source/main/utils/Utils.cpp (hunk 1 above)
Comment (Zentro) [reply source/main/utils/Utils.cpp]: This comes from the Rigs of Rods App in the Discord Developer portal.

Comment (AnimosusTheWolf) [reply source/main/utils/Utils.cpp]: @only-a-ptr You should have been added to the team a while ago.

https://discordapp.com/developers/teams

Comment (only-a-ptr) [reply source/main/utils/Utils.cpp]: Oh, apparently I am, I just didn't pay attention to it 😊

**Comment by AnotherFoxGuy** [comment]
> > @CuriousMike56 Maybe also add a in-game setting for those that don't want it? And I think the terrain name (e.g. Auriga Proving Grounds instead of 31.terrn2) should be shown instead.
> 
> @only-a-ptr Both seconded

Done in https://github.com/RigsOfRods/rigs-of-rods/pull/2412/commits/9c19e745eb3d7a2a5773e1333d56b8a4264872ad

**Comment by CuriousMike56** [comment]
👍

**Comment by CuriousMike56** [comment]
> 
> 
> @only-a-ptr RoR works fine, I can't find the status in Discord but I'll work it out.

Make sure you're using the desktop client and that `Display currently running game as status message` is enabled under Settings -> Game Activity.

**Comment by only-a-ptr** [comment]
> @CuriousMike56 Settings -> Game Activity -> Display currently running game as status message

That did the trick, thanks!

**Pull request merged**